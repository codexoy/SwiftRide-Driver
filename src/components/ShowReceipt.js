import React from "react"
import {
	StyleSheet,
	TouchableOpacity,
	Text,
	Linking,
	Alert,
	View
} from "react-native"
import { strings } from "../i18n"
import { colors } from "../consts"
import { fetchOrError } from "../actions/utils"

export default class ShowReceipt extends React.Component {
	constructor(props) {
		super(props)
		this.state = { pdf_link: "" }
	}

	_showReceipt = () => {
		Linking.openURL(this.props.item.pdf_link || this.state.pdf_link)
	}

	_createReceipt = async () => {
		const resp = await this.props.dispatch(
			fetchOrError(`/teacher/payments/${this.props.item.id}/receipt`, {
				method: "GET"
			})
		)
		if (resp) {
			this.setState({ pdf_link: resp.json["pdf_link"] }, () => {
				Alert.alert(
					strings("teacher.receipt_created_title"),
					strings("teacher.receipt_created_msg", {
						student: this.props.item.student.user.name
					}),
					[
						{
							text: strings("show_receipt_later")
						},
						{
							text: strings("show_receipt"),
							onPress: () => this._showReceipt(),
							style: "cancel"
						}
					],
					{ cancelable: false }
				)
			})
		}
	}

	render() {
		if (this.props.item.pdf_link || this.state.pdf_link) {
			return (
				<TouchableOpacity
					onPress={this._showReceipt.bind(this)}
					style={this.props.style}
				>
					<Text style={styles.receipt}>
						{strings("show_receipt")}
					</Text>
				</TouchableOpacity>
			)
		} else {
			if (this.props.user.hasOwnProperty("teacher_id")) {
				return (
					<TouchableOpacity
						onPress={this._createReceipt.bind(this)}
						style={this.props.style}
					>
						<Text style={styles.receipt}>
							{strings("teacher.create_receipt")}
						</Text>
					</TouchableOpacity>
				)
			}
		}
		return <View />
	}
}

const styles = StyleSheet.create({
	receipt: {
		color: colors.blue,
		fontWeight: "bold",
		alignSelf: "flex-start"
	}
})
