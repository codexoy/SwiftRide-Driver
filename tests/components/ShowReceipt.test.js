import React from "react"

import renderer from "react-test-renderer"

import ShowReceipt from "../../src/components/ShowReceipt"

const teacher = { name: "test", teacher_id: 1 }
const student = { name: "student", my_teacher: teacher }
const item1 = { amount: 100, details: "test", payment_type: "cash" }
const item2 = {
	amount: 100,
	details: "test",
	payment_type: "cash",
	pdf_link: "http://test.com"
}
describe("ShowReceipt", () => {
	test("view renders correctly for student", () => {
		const tree = renderer
			.create(<ShowReceipt user={student} item={item2} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test("view renders correctly for teacher", () => {
		const tree = renderer
			.create(<ShowReceipt user={teacher} item={item1} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test("view renders correctly for student without actual pdf link (should be empty view)", () => {
		const tree = renderer
			.create(<ShowReceipt user={student} item={item1} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
