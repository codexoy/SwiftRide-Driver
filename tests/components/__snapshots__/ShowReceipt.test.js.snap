// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowReceipt view renders correctly for student 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(12,116,244)",
        "fontWeight": "bold",
      }
    }
  >
    צפה בחשבונית
  </Text>
</View>
`;

exports[`ShowReceipt view renders correctly for student without actual pdf link (should be empty view) 1`] = `<View />`;

exports[`ShowReceipt view renders correctly for teacher 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(12,116,244)",
        "fontWeight": "bold",
      }
    }
  >
    צור חשבונית
  </Text>
</View>
`;
